### PR TITLE
docs(alerts & reports): add, prune, reorganize

### PR DESCRIPTION
**Pull request by sfirke**
I struggled to get my email reports set up and thought I'd propose a rewrite of that page.

### SUMMARY
- Document some additional vars, like for unauthenticated SMTP and the EMAIL_REPORTS_SUBJECT_PREFIX var.
- Rearrange for clarity.
- Copyedit, e.g., "relative" -> "relevant" and consistency of spacing and capitalization.
- Note that dev image has browser and geckodriver but non-dev does not.
- Remove some duplicative content not specific to Alerts and Reports
  - A whole `docker-compose.yml` is not necessary when there's a newer one already in the repo
  - No need to introduce Redis, Postgres components
  - The pieces of this bottom deleted block that remained possibly relevant (description of Celery beat and worker, mention of docker swarm) were moved elsewhere

I can edit to retain add'l info or reword, etc. - just let me know what to change or push changes directly to my branch.

### ADDITIONAL INFORMATION

- [X] Has associated issue: https://github.com/apache/superset/issues/20859
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/docs/installation/alerts-reports.mdx [hunk 1]
```diff
@@ -203,121 +216,11 @@ USER superset
 
 > Don't forget to set `WEBDRIVER_TYPE` and `WEBDRIVER_OPTION_ARGS` in your config if you use Chrome.
 
-### Summary of steps to turn on alerts and reporting:
```
Comment (hughhhh) [inline docs/docs/installation/alerts-reports.mdx:206]: these steps are still valid from my understanding why are we removing them?

Comment (sfirke) [reply docs/docs/installation/alerts-reports.mdx:206]: These steps aren't specific to Alerts & Reports, it's mostly a stale guide to "how to deploy with docker-compose" which is now redundant with https://superset.apache.org/docs/installation/installing-superset-using-docker-compose/ and not universally relevant here.  And it has hard-coded old content to boot, like referring to v1.0.1 and a whole docker-compose.yaml file that is out of date from https://github.com/apache/superset/blob/master/docker-compose.yml.

For instance, it starts with creating a Dockerfile and building an image - that's not necessary for Alerts & Reports.

I took the few pieces specific to Alerts & Reports or not mentioned on other docs pages and made sure they were covered above, e.g., explaining Celery worker & beat and noting that the process works in a Docker swarm environment .

**Comment by sfirke** [comment]
It looks like the recent commit from @villebro https://github.com/apache/superset/commit/a02a778cc3d0fdf2036884376b53b33444d15c03 added some new content down in the section "Old Reports feature, (version 0.38 and below)".

I will update this PR to remove the < 0.38 section entirely, no one should be configuring that now and it increases confusion like this with people reading and writing the obsolete info.  I'll work to retain the still-relevant info from that lower 0.38 section.

**Review by nytai — APPROVED** [review]
LGTM, thanks for taking this on.

**Comment by nytai** [comment]
Looks like a CI check is failing. You can run `pre-commit run --all-files` to fix

**Comment by sfirke** [comment]
Making my last updates.  @nytai is this Schedule Report section still relevant?  Or was it part of the 0.38 docs and so now obsolete?  https://superset.apache.org/docs/installation/alerts-reports/#schedule-reports

**Comment by sfirke** [comment]
FYI it said "merge conflict" and then when I went to resolve, 0 conflicts (as expected).  So I clicked commit merge 🤷

**Pull request merged**